includes/class-reader-activation.php

```diff
@@ -0,0 +1,328 @@
+<?php
+/**
+ * Reader Activation.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Reader Activation Class.
+ */
+final class Reader_Activation {
+
+	const AUTH_INTENTION_COOKIE = 'np_auth_intention';
+
+	/**
+	 * Reader user meta keys.
+	 */
+	const READER         = 'np_reader';
+	const EMAIL_VERIFIED = 'np_reader_email_verified';
+
+	/**
+	 * Initialize hooks.
+	 */
+	public static function init() {
+		if ( self::is_enabled() ) {
+			\add_action( 'clear_auth_cookie', [ __CLASS__, 'clear_auth_intention_cookie' ] );
+			\add_filter( 'login_form_defaults', [ __CLASS__, 'add_auth_intention_to_login_form' ], 20 );
+			\add_action( 'resetpass_form', [ __CLASS__, 'verify_reader_email' ] );
```

Comment:
b310d1744937a1bf69dd682e46da993d68204324 adds a hook to the `reset_password`. I preserved the `resetpass_form` hook because the reader may access the page and never click to actually reset the password. We don't really care about that, we just want to know if the reader clicked the link.

The email flow that WooCommerce provides is a bit different than what we'd like to achieve with this first setup. It forces the reader to go through the "lost password" flow in order to set the password for the first time.

As we move forward we may want to control/take over this email sent from WooCommerce to provide a password creation link right at reader registration, as WP's default email does.

A different PR will tackle how we handle emails in an integrated way and we can continue this discussion there.